src/bootstrap/compile.rs

```diff
@@ -150,6 +150,11 @@ pub fn std_cargo(builder: &Builder,
         cargo.env("MACOSX_DEPLOYMENT_TARGET", target);
     }
 
+    // FIXME: Temporary detection of SJLJ MinGW compilers.
+    if build.build.build.contains("linux") && target == "i686-pc-windows-gnu" {
```

Comment:
Should this be `builder.build.build`?

Reply:
target.contains("linux") ?

as like:
...
766 !target.contains("freebsd") &&
767 !target.contains("windows") &&
768 !target.contains("apple") {
...